bigc/api_client.py

```diff
@@ -51,21 +73,41 @@ def delete(self, *args, **kwargs):
         return self.request('DELETE', *args, **kwargs)
 
     @abstractmethod
-    def get_many(self, path: str, *, page_size: Optional[int] = None, **kwargs) -> Generator:
+    def get_many(
+        self,
+        path: str,
+        *,
+        page_size: int | None = None,
+        params: dict[str, Any] | None = None,
+        timeout: float | None = None,
+    ) -> Iterator[Any]:
         """Make a request to a paginated BigCommerce API endpoint"""
         pass
 
     @abstractmethod
     def _prepare_url(self, path: str) -> str:
         pass
 
-    def _get_standard_request_headers(self) -> dict:
+    def _get_standard_request_headers(self) -> dict[str, str]:
         return {
             'Accept': 'application/json',
             'Content-Type': 'application/json',
             'X-Auth-Token': self.access_token,
         }
 
+    @staticmethod
+    def _process_params(params: dict[str, Any] | None) -> dict[str, Any] | None:
```

Comment:
If we're pre-processing params, then I think would make more sense if the return type was `dict[str, str] | None` and we stringified all the params here, rather than relying on requests to do it.